### ISSUE TITLE
hints for cnf solver

### ISSUE THREAD

**Issue by sbelodon** [arminbiere/cadical#39]
Hello,

the situation is following:
I've got cnf formula after Tseytin transofrmation.
It relies on 32 variables.
but Tseytin tranformation introduces much more intermidiate variables(~300 000) (we don't need they values)
I'm want to know your opinion about 2 points:
1.If solver will know that variables 0-31 for example is base(others are intermidiate) and take only it for decisions (except pure literals and other 100% vars) will it speed up solving?
2 will hints from Tseytin transform like variable 3 is conjuction (or other function) of var 1 and 2 speed up solving?

Thanks for answer in advance

**Comment by arminbiere** [comment]
Also let me mention that it is more tricky to have such a function in CaDiCaL or Kissat which have multiple decision queues, one actually being a doubly linked list without a sorting function.  For the VMTF doubly linked list one would need two queues (one with higher and one with lower priority).  This is awkward (adds lots of code) and probably slow.  Then for Tseitin transformation, where the internal variables are functionally dependent on the inputs, you really only need the inputs in the decision queue (since the others are set through propagation).  As soon you do for instance Plaisted-Greenbaum or some fancy preprocessing this breaks down though.

**Comment by violetcodet** [comment]
the same problem
it's really hard to model count when the number of variables is large （100 key variables and 300000 ~ intermidiate variables）.
Can you give me some hints if you have solve this problem.

**Comment by arminbiere** [comment]
Not really, but we do have a 'structural' model counter which uses 'dual reasoning' exactly for this reason.  It is called 'dualiza' and can be fond here: [https://github.com/arminbiere/dualiza](https://github.com/arminbiere/dualiza)
